apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.BenchmarkingTool/Program.cs

```diff
@@ -0,0 +1,244 @@
+            long bytesLeft = configuration.DataSizeInBytes;
+            byte[] buffer = new byte[1024 * 1024];
+            using var output = File.Create(configuration.LocalFile);
+            while (bytesLeft > 0)
+            {
+                random.NextBytes(buffer);
+                int count = (int) Math.Min(bytesLeft, buffer.Length);
+                output.Write(buffer, 0, count);
+                bytesLeft -= count;
+            }
+        }
+
+        private static void Write(Configuration configuration, StorageClient client, PerformanceLogger logger, string opName)
+        {
+            string elapsedTimeUs = "-";
+            bool isSucccess = false;
+
+            var logs = new string[10]; //logs array should be in same order as s_logHeaders.
+            try
+            {
+                CreateRandomFile(configuration);
+
+                UploadObjectOptions options = null;
+
+                if (configuration.UploadChunkSizeInBytes != 0)
+                {
+                    options = new UploadObjectOptions { ChunkSize = configuration.UploadChunkSizeInBytes };
+                }
+
+                long totalElapsedMicroseconds = 0;
+
+
+                logs[0] = opName;
+                logs[1] = configuration.DataSizeInBytes.ToString(CultureInfo.InvariantCulture);
+                logs[2] = "NA";
+                logs[3] = options != null ? options.ChunkSize.ToString() : ResumableUpload.DefaultChunkSize.ToString();
+                logs[4] = "TRUE";
+                logs[5] = "NA";
+                logs[6] = "JSON";
+                logs[7] = "-1";
+
+                Console.WriteLine("Uploading..");
+                using (var input = File.OpenRead(configuration.LocalFile))
+                {
+                    BufferedStream bs = new BufferedStream(input);
+                    var startTime = s_stopwatch.Elapsed;
+                    client.UploadObject(configuration.Bucket, configuration.ObjectName, "application/binary", input, options);
+
+                    var elapsedTimespan = s_stopwatch.Elapsed - startTime;
+                    totalElapsedMicroseconds = elapsedTimespan.Ticks / TicksPerMicrosecond;
+                }
+
+                elapsedTimeUs = totalElapsedMicroseconds.ToString(CultureInfo.InvariantCulture);
+                isSucccess = true;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Error while upload : {e.Message}");
+            }
+            finally
+            {
+                logs[8] = elapsedTimeUs;
+                logs[9] = isSucccess ? "OK" : "FAIL";
+                logger.Log(logs);
+
+                if (configuration != null & File.Exists(configuration.LocalFile))
+                {
+                    File.Delete(configuration.LocalFile);
+                }
+            }
+        }
+
+        private static void Read(Configuration configuration, StorageClient client, PerformanceLogger logger, string opName)
```

Comment:
Same comments as before, basically - but let's get the write test into a good state, then make this one look the same.